docs/query-cascade.md

```
@@ -0,0 +1,256 @@
+| `originator_peer` | yes | ✓ | ✓ | Peer that started the query (for routing) |
+| `is_response` | yes | `False` | `True` | Distinguishes request from response |
+| `responder_peer` | — | — | ✓ | Peer that generated this answer |
+| `responder_site_id` | — | — | ✓ | Site ID of the responder |
+
+---
+
+## QUERY Algorithm
+
+### Request Flow (is_response=False)
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)